dagger/dsl/build.py

```diff
@@ -38,6 +39,7 @@ def build(dag: NodeInvocationRecorder) -> DAG:
 
 def _build(
     build_func: Callable,
+    has_a_parent: bool,
```

Comment:
Yeah, but I feel it's a bit awkward to have 2 separate arguments:

```python
has_a_parent: bool,
inputs_from_parent: Mapping[str, NodeInputReference],
```

Because the API allows you to pass `(has_a_parent=False, inputs_from_parent={"a": ParameterUsage(...)})`.

I'm thinking now that it may be better to have:

```python
parent: Optional[ParentMetadata] = None,
```

Or something like this. Thoughts?

Reply:
True, I would name it `parent_metadata` or even more verbose `parent_metadata_if_parent_exists`